x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/googlevertexai/GoogleVertexAiRequest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.googlevertexai;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+
+import org.apache.http.client.methods.HttpPost;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.xpack.inference.external.request.Request;
+import org.elasticsearch.xpack.inference.services.googlevertexai.GoogleVertexAiSecretSettings;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.xpack.inference.external.request.RequestUtils.createAuthBearerHeader;
+
+public interface GoogleVertexAiRequest extends Request {
+    List<String> AUTH_SCOPE = Collections.singletonList("https://www.googleapis.com/auth/cloud-platform");
+
+    static void decorateWithBearerToken(HttpPost httpPost, GoogleVertexAiSecretSettings secretSettings) {
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
```

Comment:
Yes, good point! Adjusted with [Use SocketAccess instead of AccessController directly](https://github.com/elastic/elasticsearch/pull/110090/commits/e3a8791231037da3f5013cf055b310e4a0a155b0)

Reply:
fyi: I've adjusted the change to use `AccessController` again and add an explicit `SpecialPermission.check()` with [Use AccessController and add special permission check](https://github.com/elastic/elasticsearch/pull/110090/commits/adf84b7971a8bdf812829655bd381a016c6a5b2a).

For some reason I get a permission denied error again using `SocketAccess`, which is confusing as now the code is basically the same and works? Looking through the codebase I see a lot of replications of the `SocketAccess` class, maybe it has something to do with some module magic I'm not aware of? Appreciating any hints!

I suggest we keep it like this for now, if we don't find a solution soonish. I'll create an issue and revisit it afterwards.

Reply:
Could it be the case, because the `x-pack core` plugin misses `permission java.lang.RuntimePermission "accessDeclaredMembers"`, which `GoogleCredentials` needs and the security manager uses the intersection of the two permission sets of the `core` and `inference` plugin?

Reply:
Ah ok, leaving it like you have is fine 👍